delta/imagery/rectangle.py

```diff
@@ -199,50 +200,101 @@ def make_tile_rois(self, tile_shape, overlap_shape=(0, 0), include_partials=True
+                if r > 0:
+                    unique_tile.min_y += half_overlap[1]
+                if c < num_tiles[0]-1:
+                    unique_tile.max_x -= half_overlap[0]
+                if r < num_tiles[1]-1:
+                    unique_tile.max_y -= half_overlap[1]
+                unique_tile = unique_tile.get_intersection(self)
+
                 if include_partials: # Crop the tile to the valid area and use it
                     tile = tile.get_intersection(self)
                     if tile.width() < min_width or tile.height() < min_height:
                         continue
                 else: # Only use it if the uncropped tile fits entirely in this Rectangle
-                    if not self.contains_rect(tile):
+                    if not containing_rect.contains_rect(tile):
                         if not partials_overlap:
                             continue
-                        tile = Rectangle(min(self.max_x, tile.max_x) - tile_width,
-                                         min(self.max_y, tile.max_y) - tile_height,
+                        # Try shifting tile "back" in x/y so that we can fit the entire proposed tile
+                        new_max_x = tile.max_x
+                        new_max_y = tile.max_y
+                        if tile.max_x > containing_rect.max_x:
+                            new_max_x = min(containing_rect.max_x, tile.max_x)
+                        if tile.max_y > containing_rect.max_y:
+                            new_max_y = min(containing_rect.max_y, tile.max_y)
+
+                        tile = Rectangle(new_max_x - tile_width, new_max_y - tile_height,
                                          width=tile_width, height=tile_height)
-                        if not self.contains_rect(tile):
+                        if not containing_rect.contains_rect(tile):
                             continue
                 if by_block:
                     row_tiles.append(tile)
+                    unique_row_tiles.append(unique_tile)
                 else:
                     output_tiles.append(tile)
+                    unique_tiles.append(unique_tile)
 
             if by_block and row_tiles:
-                row_rect = Rectangle(row_tiles[0].min_x, row_tiles[0].min_y, row_tiles[-1].max_x, row_tiles[-1].max_y)
+                row_rect = Rectangle(row_tiles[0].min_x, row_tiles[0].min_y,
+                                     row_tiles[-1].max_x, row_tiles[-1].max_y)
                 for r in row_tiles:
                     r.shift(-row_rect.min_x, -row_rect.min_y)
+                for r in unique_row_tiles:
+                    r.shift(-row_rect.min_x, -row_rect.min_y)
                 output_tiles.append((row_rect, row_tiles))
+                unique_tiles.append((row_rect, unique_row_tiles))
+
+        return output_tiles, unique_tiles
 
-        return output_tiles
+    def make_tile_rois_yx(self, tile_shape, overlap_shape=(0, 0), include_partials=True, min_shape=(0, 0),
```

Comment:
I kept the x, y ordering for the rest of the Rectangle class since it is not explicitly for images and I did not want to have one function "silently" change the order.  This one allows us to use the more convenient order while making it harder to get confused about what the order is.